### PR TITLE
Add org.apache.directory.studio

### PR DESCRIPTION
**Pull request by wjt**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://issues.apache.org/jira/browse/DIRSTUDIO-1153?focusedCommentId=17511282&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17511282
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
  - I do not own the domain, but the app ID is based on the upstream site.
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
  - I have not yet submitted the .desktop or metainfo files upstream, though I have mentioned them in the ticket about a Flatpak package linked above.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: :-( I thought Java was "run everywhere"

**Comment by wjt** [comment]
I'm sure a sourceful build would be possible (gaining aarch64 support) but I'm afraid the limit of my interest in this app is to be able to create LDAP accounts for new team members at Endless OS Foundation, on Endless OS, with a UI. The easiest way to run it was to make a Flatpak with the OpenJDK extension, so the marginal cost of adding the `.desktop` and `.metainfo.xml` files was quite low.

File: org.apache.directory.Studio.desktop [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Type=Application
+Name=Apache Directory Studio
+Comment=A complete directory tooling platform, intended to be used with any LDAP server
+Icon=org.apache.directory.Studio
+Exec=/app/ApacheDirectoryStudio/ApacheDirectoryStudio
```
Comment (hfiguiere) [inline org.apache.directory.Studio.desktop:6]: you should have a symlink in `/app/bin` and just put that (no path)

File: org.apache.directory.Studio.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+app-id: org.apache.directory.Studio
+
+sdk: org.gnome.Sdk
+runtime: org.gnome.Platform
```
Comment (hfiguiere) [inline org.apache.directory.Studio.yaml:4]: does this really need the GNOME platform?

File: org.apache.directory.Studio.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+runtime-version: '42'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+
+command: /app/ApacheDirectoryStudio/ApacheDirectoryStudio
+finish-args:
+  - --device=dri
+  - --device=shm
```
Comment (hfiguiere) [inline org.apache.directory.Studio.yaml:12]: Are you sure it needs that?

File: org.apache.directory.Studio.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  - name: ApacheDirectoryStudio
+    buildsystem: simple
+    build-commands:
+      - install -Dt /app/share/applications org.apache.directory.Studio.desktop
+      - install -Dt /app/share/metainfo org.apache.directory.Studio.metainfo.xml
+      - install -D studio-icon_128x128.png /app/share/icons/hicolor/128x128/apps/org.apache.directory.Studio.png
+    sources:
+      - type: file
+        path: org.apache.directory.Studio.desktop
```
Comment (hfiguiere) [inline org.apache.directory.Studio.yaml:50]: so Apache ship pre builds for Linux and don't even provide a .desktop file nor an icon?

File: org.apache.directory.Studio.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
+app-id: org.apache.directory.Studio
+
+sdk: org.gnome.Sdk
+runtime: org.gnome.Platform
+runtime-version: '42'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+
+command: /app/ApacheDirectoryStudio/ApacheDirectoryStudio
```
Comment (hfiguiere) [inline org.apache.directory.Studio.yaml:9]: like for the desktop. You should have a symlink in `/app/bin` and use the command name, no path.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: flathub.json (hunk 1 above)
Comment (wjt) [reply flathub.json:2]: It uses (and bundles) [SWT](https://en.wikipedia.org/wiki/Standard_Widget_Toolkit) which links to GTK.

File: org.apache.directory.Studio.yaml (hunk 5 above)
Comment (wjt) [reply org.apache.directory.Studio.yaml:50]: I couldn't find a `.desktop` file in the bundle.

There is a 48×48 XPM at the top level of the bundle… Oh, there actually is a 128×128 PNG buried in there.

File: org.apache.directory.Studio.yaml (hunk 4 above)
Comment (wjt) [reply org.apache.directory.Studio.yaml:12]: I'm not!

File: org.apache.directory.Studio.yaml (hunk 6 above)
Comment (wjt) [reply org.apache.directory.Studio.yaml:9]: I swear I tried this and it didn't work, but now it seems to.

File: org.apache.directory.Studio.yaml (hunk 3 above)
Comment (wjt) [reply org.apache.directory.Studio.yaml]: Seems you're right.

File: flathub.json (hunk 1 above)
Comment (wjt) [reply flathub.json:2]: It doesn't look like a build from source would be particularly difficult, assuming the existence of a maven manifest generator or some manual work to fake it… But I'm afraid I'm not about to work on it.

https://gitbox.apache.org/repos/asf?p=directory-studio.git;a=blob;f=build.sh;h=698c8e80d4a89c1aeddef7794831c7b3a39b359f;hb=HEAD

**Comment by barthalion** [comment]
bot, build org.apache.directory.Studio

**Comment by wjt** [comment]
```
<groupId>org.apache.directory.studio</groupId>
```

looks like I should lowercase it.

**Comment by barthalion** [comment]
I don't think it particularly matters if the application does not use D-Bus, but I will leave the decision to you.

**Comment by nedrichards** [comment]
I think it's likely to ease any upstream adoption of app-id named assets to have it actually match what they think the app-id is so worth doing.

File: org.apache.directory.Studio.yaml (hunk 3 above)
Comment (wjt) [reply org.apache.directory.Studio.yaml]: It actually does need the GNOME Platform for webkit reasons.

```
SWT Webkit: Warning, You are using an old version of webkitgtk. (pre 2.4) BrowserFunction functionality will not be avaliable
SWT WebKit: error initializing DBus server, dBusServer == 0
SWT.CHROMIUM style is not supported in Wayland.
```

With this change the welcome page does not work.

File: org.apache.directory.studio.yaml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+command: ApacheDirectoryStudio
+finish-args:
+  - --device=dri
+  - --env=PATH=/app/bin:/app/jre/bin:/usr/bin
+  - --persist=.ApacheDirectoryStudio
+  - --persist=.eclipse
+  - --share=ipc
+  - --share=network
+  - --socket=wayland
+  - --socket=fallback-x11
```
Comment (wjt) [inline org.apache.directory.studio.yaml:18]: FWIW while the app does *work* with Wayland, it's definitely buggier than with X11. For instance, the "are you sure you want to quit?" dialog is unusably small when first opened.

However I think it's probably good enough.

**Comment by barthalion** [comment]
bot, build org.apache.directory.studio

**Pull request closed without merging**